### ISSUE TITLE
Loading KeepassHTTP results in empty error message

### ISSUE THREAD

**Issue by aheusingfeld** [pfn/keepasshttp#121]
I'm currently running Keepass 2.22 on CrossOver 12.2.1.26977 on Mac OS X 10.8.4. Crossover is a customized Wine

To access keepass in Firefox I
- downloaded KeepassHTTP from https://github.com/pfn/keepasshttp (to be exact, this is the version 1.7.1.0 I used https://github.com/pfn/keepasshttp/blob/c4ff23048c1b30be55b496628f6fab2c
  bc805071/KeePassHttp.plgx)
- dropped it into the Keepass directory in ~/Library/Application Support/CrossOver/Bottles/KeePass2/drive_c/Program Files/KeePass Password Safe 2/
- started the Keepass2.app previously created by CrossOver

While starting Keepass tries to load the plugin but it fails with an empty popup error dialog. Empty in this case means there is no message in the dialog, just the OK button. 

The same happens when I exchange the "KeePassHttp.pglx" file with "Newtonsoft.Json.dll" and "KeePassHttp.dll" from the [mono directory](https://github.com/pfn/keepasshttp/tree/master/mono). 

Have you got an idea what the cause might be or do you know whether Keepass write a logfile about such load time errors?

BTW: Can you tell me which port KeePassHTTP tries to occupy? I'd like to check whether this port is already in use. 

Thanks in advance for your support.

NOTE: I posted the same problem description as a support request at Codeweavers (https://www.codeweavers.com/compatibility/browse/name/?app_id=1641;forum=1;msg=146314)

**Comment by boky8** [comment]
I can confirm that I can repeat this error in 2015.

Running KeePass 2.26 with the latest KeePassHttp will create a blank popup message. KeePassHttp seems to be installed semi-installed though (you can see and use the "KeePassHttp Options..." from the "Tools" menu) but you can't see it on the "Plugins..." page.

Also the listening port does not open.

There's nothing in the Wine logs.


Anyway to crack up debbuging of KeePassHttp to stdout so we can see what's happening and paste the logs here?

**Comment by Starli0n** [comment]
I have the exact same issue in 2016
Btw the default port is 19455.
Tools > KeePassHtpp Options... > Advanced

The plugin list is empty
Tools > Plugins...

It seems that the plugin KeePassHtpp is not loaded correctly whereas the same configuration works correctly on a native Windows + KeePassHtpp + PassIFox

**Comment by jasonxh** [comment]
This is due to the use of `System.Net.HttpListener`, which depends on `httpapi.dll` 2.0, which is not available in Wine. I found a workaround to replace this dependency completely with a managed implementation. If anyone's still trying to get KeePassHttp to work with Wine, you can checkout my forked repo at <https://github.com/jasonxh/keepasshttp>.

**Comment by pfn** [comment]
Actually, if the managed implemention of the HTTP listener is fairly
robust, I would appreciate a PR that would switch to this.

**Comment by jasonxh** [comment]
@pfn yeah it's been fairly stable for me for more than a month. I've opened PR https://github.com/pfn/keepasshttp/pull/289

**Comment by nifr** [comment]
Nice PR @jasonxh - thanks for the contribution!

**Comment by oit63** [comment]
@pfn @jasonxh thanks, works fine on macOS 10.14.5
